crates/nostr/src/event/builder.rs

```diff
@@ -30,11 +34,12 @@ pub enum Error {
     #[error(transparent)]
     Key(#[from] key::Error),
     /// Secp256k1 error
-    #[error(transparent)]
-    Secp256k1(#[from] secp256k1::Error),
+
+    #[error("Secp256k1 Error: {0}")]
+    Secp256k1(secp256k1::Error),
     /// JSON error
-    #[error(transparent)]
-    Json(#[from] serde_json::Error),
+    #[error("Serde json Error: {0}")]
+    Json(serde_json::Error),
```

Comment:
Yeah, the problem is the following:

`thiserror` crate assumes that the enum/struct implements the `std::error::Error` trait, which is not available in `no_std`. However the `thiserror-core` [crate](https://crates.io/crates/thiserror-core) switched to the `core::error::Error` trait, [which is available on `nightly`](https://crates.io/crates/thiserror-core). 

If you are not going to use the `thiserror` heavily, then I would say removing it would be a good idea.

Reply:
I think we can remove `thiserror` dep.

Maybe is better to do that in another PR.
If you don't have enough time, I can do it in the next days.

Reply:
I would appreciate it if you did that in a another PR, thanks!